mlflow/models/evaluation/default_evaluator.py

```diff
@@ -494,28 +499,49 @@ def _log_model_explainability(self):
             f: f2 for f, f2 in zip(self.feature_names, truncated_feature_names)
         }
 
-        sampled_X = shap.sample(self.X, sample_rows)
+        sampled_X = shap.sample(self.X, sample_rows, random_state=0)
+        if algorithm == "kernel":
+            background_X = shap.sample(self.X, sample_rows, random_state=3)
+        else:
+            background_X = None
 
         if isinstance(sampled_X, pd.DataFrame):
             # For some shap explainer, the plot will use the DataFrame column names instead of
             # using feature_names argument value. So rename the dataframe column names.
             sampled_X = sampled_X.rename(columns=truncated_feature_name_map, copy=False)
+            if background_X is not None:
+                background_X = background_X.rename(columns=truncated_feature_name_map, copy=False)
+        else:
+            # If sample_X is numpy array, convert to pandas dataframe.
+            sampled_X = pd.DataFrame(sampled_X, columns=truncated_feature_name_map)
+            background_X = pd.DataFrame(background_X, columns=truncated_feature_name_map)
 
         if algorithm:
-            supported_algos = ["exact", "permutation", "partition"]
+            supported_algos = ["exact", "permutation", "partition", "kernel"]
             if algorithm not in supported_algos:
                 raise ValueError(
                     f"Specified explainer algorithm {algorithm} is unsupported. Currently only "
                     f"support {','.join(supported_algos)} algorithms."
                 )
-            explainer = shap.Explainer(
-                self.predict_fn,
-                sampled_X,
-                feature_names=truncated_feature_names,
-                algorithm=algorithm,
-            )
+            if algorithm == "kernel":
+                # predict_fn = lambda x: self.predict_fn(pd.DataFrame(x, columns=sampled_X.columns))
+                mode = sampled_X.mode().iloc[0]
+                sampled_X = sampled_X.fillna(mode)
+                background_X = background_X.fillna(mode)
+                explainer = shap.KernelExplainer(self.predict_fn, background_X, link="identity")
```

Comment:
The link function has other options, shall we add config for this ?

Reply:
I would definitely add the ability to configure this to use `logit`. `identity` is useless with LogisticRegression or any other model that relies on probability thresholds as their output.